mlflow/server/js/src/components/RunsTableColumnSelectionDropdown.js

```diff
@@ -0,0 +1,152 @@
+      })),
+    ];
+
+    // Parameters
+    if (paramKeyList.length > 0) {
+      data.push({
+        title: 'Parameters',
+        key: ColumnTypes.PARAMS,
+        children: paramKeyList.map((key) => ({ key: `${ColumnTypes.PARAMS}-${key}`, title: key })),
+      });
+    }
+
+    // Metrics
+    if (metricKeyList.length > 0) {
+      data.push({
+        title: 'Metrics',
+        key: ColumnTypes.METRICS,
+        children: metricKeyList.map((key) => ({
+          key: `${ColumnTypes.METRICS}-${key}`,
+          title: key,
+        })),
+      });
+    }
+
+    // Tags
+    if (visibleTagKeyList.length > 0) {
+      data.push({
+        title: 'Tags',
+        key: ColumnTypes.TAGS,
+        children: visibleTagKeyList.map((key) => ({
+          key: `${ColumnTypes.TAGS}-${key}`,
+          title: key,
+        })),
+      });
+    }
+
+    return data;
+  }
+
+  getCheckedKeys() {
+    const { paramKeyList, metricKeyList, visibleTagKeyList, categorizedUncheckedKeys } = this.props;
+    return [
+      ..._.difference(
+        Object.values(ExperimentViewUtil.AttributeColumnLabels),
+        categorizedUncheckedKeys[ColumnTypes.ATTRIBUTES],
+      ).map((key) => `${ColumnTypes.ATTRIBUTES}-${key}`),
+      ..._.difference(paramKeyList, categorizedUncheckedKeys[ColumnTypes.PARAMS])
+        .map((key) => `${ColumnTypes.PARAMS}-${key}`),
+      ..._.difference(metricKeyList, categorizedUncheckedKeys[ColumnTypes.METRICS])
+        .map((key) => `${ColumnTypes.METRICS}-${key}`),
+      ..._.difference(visibleTagKeyList, categorizedUncheckedKeys[ColumnTypes.TAGS])
+        .map((key) => `${ColumnTypes.TAGS}-${key}`),
+    ];
+  }
+
+  handleSearchInputEscapeKeyPress = () => {
+    this.setState({ menuVisible: false });
+  };
+
+  handleVisibleChange = (menuVisible) => {
+    this.setState({ menuVisible });
+  };
+
+  render() {
+    const { menuVisible } = this.state;
+    const content = (
+      <Menu style={{ maxHeight: 480, overflowY: 'scroll' }}>
+        <SearchTree
+          data={this.getData()}
+          onCheck={this.handleCheck}
+          checkedKeys={this.getCheckedKeys()}
+          onSearchInputEscapeKeyPress={this.handleSearchInputEscapeKeyPress}
+        />
+      </Menu>
+    );
+    return (
+      <Dropdown
+        overlay={content}
+        trigger={['click']}
+        visible={menuVisible}
+        onVisibleChange={this.handleVisibleChange}
+      >
+        <Button style={{ height: 34, display: 'flex', alignItems: 'center' }}>
+          <Icon type='setting' style={{ marginTop: 2 }} />
+          Columns
+        </Button>
+      </Dropdown>
+    );
+  }
+}
+
+function getCategorizedUncheckedKeys(checkedKeys, allKeys) {
+  const uncheckedKeys = _.difference(allKeys, checkedKeys);
+  const result = {
+    [ColumnTypes.ATTRIBUTES]: [],
+    [ColumnTypes.PARAMS]: [],
+    [ColumnTypes.METRICS]: [],
+    [ColumnTypes.TAGS]: [],
+  };
+  uncheckedKeys.forEach((key) => {
+    const [columnType, rawKey] = key.split('-');
```

Comment:
This line doesn't split `key` properly when a metric (or parameter) name contains `-`.

Reply:
Tested the code below and it worked.

```diff
+    const [columnType, rawKey] = key.split('-');
-    const [columnType, rawKey] = key.split(/-(.+)/);
```

Ref. : https://stackoverflow.com/questions/4607745/split-string-only-on-first-instance-of-specified-character

Reply:
Fixed